src/main.ts

```diff
@@ -102,34 +106,44 @@ async function runTrivy(): Promise<number> {
     };
 
     const trivyToolRunner = new ToolRunner(trivyPath, [imageName], trivyOptions);
-    const trivyStatus = await trivyToolRunner.exec();
+    const trivyStatus = await trivyToolRunner.exec();    
     return trivyStatus;
 }
 
-async function runDockle(): Promise<void> {
-    const docklePath = await getDockle();
-    const dockleEnv = await setDockleEnvVariables();
+async function runDockle(): Promise<number> {
+    const docklePath = await dockleHelper.getDockle();
+    const dockleEnv = await getDockleEnvVariables();
     const imageName = inputHelper.imageName;
 
     const dockleOptions: ExecOptions = {
         env: dockleEnv,
         ignoreReturnCode: true
     };
 
-    const dockleToolRunner = new ToolRunner(docklePath, [imageName], dockleOptions);
-    await dockleToolRunner.exec();
+    let dockleArgs = [ '-f', 'json', '-o', fileHelper.getDockleOutputPath(), '--exit-code', dockleHelper.DOCKLE_EXIT_CODE.toString(), imageName ];
+    const dockleToolRunner = new ToolRunner(docklePath, dockleArgs, dockleOptions);
+    const dockleStatus =  await dockleToolRunner.exec();
+    return dockleStatus;
 }
 
 async function run(): Promise<void> {
     const trivyStatus = await runTrivy();
-    const addCISChecks = inputHelper.addCISChecks;
-    if (addCISChecks.toLowerCase() === "true") {
-        await runDockle();
+    let dockleStatus: number;
+    if (dockleHelper.isCisChecksEnabled()) {
+        dockleStatus = await runDockle();
+    }
+
+    try {
+        console.log("Creating check run...");
+        await gitHubHelper.createCheckRunWithScanResult(trivyStatus, dockleStatus);
```

Comment:
Construct the content here instead.

Reply:
Or fetch it from the respective helpers. Adding the logic of creation of Content to GitHub doesn't make sense

Reply:
Refactored as discussed.